sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -716,18 +716,22 @@ size_t coap_opt_put_string(uint8_t *buf, uint16_t lastonum, uint16_t optnum,
 static ssize_t _add_opt_pkt(coap_pkt_t *pkt, uint16_t optnum, uint8_t *val,
                             size_t val_len)
 {
-    assert(pkt->options_len < NANOCOAP_NOPTS_MAX);
-
     uint16_t lastonum = (pkt->options_len)
             ? pkt->options[pkt->options_len - 1].opt_num : 0;
     assert(optnum >= lastonum);
 
+    if (pkt->options_len >= NANOCOAP_NOPTS_MAX) {
```

Comment:
why not keep this check above (and hence first when entering the function)?

Reply:
I believe I wanted to handle the assert first, but it's not a problem to move up.